packages/template-retail-react-app/app/components/shopper-agent/index.jsx

```diff
@@ -0,0 +1,223 @@
+        'scriptSourceUrl',
+        'scrt2Url',
+        'salesforceOrgId',
+        'commerceOrgId',
+        'siteId'
+    ]
+
+    const isValid = requiredFields.every((key) => typeof commerceAgent[key] === 'string')
+    if (!isValid) {
+        console.error('Invalid commerce agent settings.')
+    }
+    return isValid
+}
+
+function isEnabled(enabled) {
+    return enabled === 'true' && onClient
+}
+
+function ShopperAgentWindow({commerceAgentConfiguration, locale, basketId}) {
+    const theme = useTheme()
+    const {
+        embeddedServiceName,
+        embeddedServiceEndpoint,
+        scriptSourceUrl,
+        scrt2Url,
+        salesforceOrgId,
+        commerceOrgId,
+        siteId
+    } = commerceAgentConfiguration
+
+    const {usid} = useUsid()
+
+    useEffect(() => {
+        const handleEmbeddedMessagingReady = () => {
+            window.embeddedservice_bootstrap.prechatAPI.setHiddenPrechatFields({
+                SiteId: siteId,
+                Locale: locale,
+                OrganizationId: commerceOrgId,
+                UsId: usid,
+                IsCartMgmtSupported: true
+            })
+        }
+
+        const handleEmbeddedMessagingWindowMaximized = () => {
+            const zIndex = theme.zIndices.sticky + 1
+            const embeddedMessagingFrame = document.body.querySelector(
+                'div.embedded-messaging iframe'
+            )
+            if (embeddedMessagingFrame) {
+                embeddedMessagingFrame.style.zIndex = zIndex
+            }
+        }
+
+        window.addEventListener('onEmbeddedMessagingReady', handleEmbeddedMessagingReady)
+        window.addEventListener(
+            'onEmbeddedMessagingWindowMaximized',
+            handleEmbeddedMessagingWindowMaximized
+        )
+
+        // Cleanup function
+        return () => {
+            window.removeEventListener('onEmbeddedMessagingReady', handleEmbeddedMessagingReady)
+            window.removeEventListener(
+                'onEmbeddedMessagingWindowMaximized',
+                handleEmbeddedMessagingWindowMaximized
+            )
+        }
+    }, [commerceAgentConfiguration, usid, theme.zIndices.sticky])
+
+    // whenever the basketId changes, update the hidden prechat fields
+    useEffect(() => {
+        const handleEmbeddedMessagingButtonClicked = () => {
+            window.embeddedservice_bootstrap.prechatAPI.setHiddenPrechatFields({
+                BasketId: basketId
+            })
+        }
+
+        window.addEventListener(
+            'onEmbeddedMessagingButtonClicked',
+            handleEmbeddedMessagingButtonClicked
+        )
+
+        // Cleanup function
+        return () => {
+            window.removeEventListener(
+                'onEmbeddedMessagingButtonClicked',
+                handleEmbeddedMessagingButtonClicked
+            )
+        }
+    }, [basketId])
+
+    // Load the embedded messaging script
+    const scriptLoadStatus = useScript(scriptSourceUrl)
```

Comment:
what is the reason this script needs to be loaded this way? Since the url is configurable, how do we know if it's secured and trusted?

Reply:
In defualt.js.hbs, the value is sourced form process.env.COMMERCE_AGENT_SETTINGS - so as long as someone specifies it in the environment setting, it can be overriden to anything they want, no?

I looked at the validate function, it only checks for requiredness, where is the code to ensure it comes from salesforce.com?